utils/inpsSoapClient.ts

```diff
@@ -0,0 +1,110 @@
+import { format } from "date-fns";
+import { fromNullable, toError } from "fp-ts/lib/Either";
+import { TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import fetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import {
+  ConsultazioneSogliaIndicatoreInput,
+  FornituraNucleoEnum
+} from "../generated/definitions/ConsultazioneSogliaIndicatoreInput";
+import { ConsultazioneSogliaIndicatoreResponse } from "../generated/definitions/ConsultazioneSogliaIndicatoreResponse";
+
+const getSOAPRequest = (
+  dataValidita: string,
+  fiscalCode: string,
+  fornituraNucleo: FornituraNucleoEnum,
+  codiceSoglia: string
+) => `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://inps.it/ConsultazioneISEE">
+<soapenv:Header/>
+<soapenv:Body>
+  <con:ConsultazioneSogliaIndicatoreRequest>
+    <con:Richiesta DataValidita="${dataValidita}" CodiceFiscale="${fiscalCode}" FornituraNucleo="${fornituraNucleo}" CodiceSoglia="${codiceSoglia}"/>
+  </con:ConsultazioneSogliaIndicatoreRequest>
+</soapenv:Body>
+</soapenv:Envelope>`;
+
+const INPS_NAMESPACE = "http://inps.it/ConsultazioneISEE";
+
+export interface ISoapClientAsync {
+  ConsultazioneSogliaIndicatore: (
+    params: ConsultazioneSogliaIndicatoreInput
+  ) => TaskEither<Error, ConsultazioneSogliaIndicatoreResponse>;
+}
+
+export function createClient(
+  INPS_SERVICE_HOST: NonEmptyString,
+  endpoint: NonEmptyString
+): ISoapClientAsync {
+  return {
+    ConsultazioneSogliaIndicatore: (
+      params: ConsultazioneSogliaIndicatoreInput
+    ) => {
+      return tryCatch(async () => {
+        const requestPayload = getSOAPRequest(
+          params.DataValidita || format(new Date(), "yyyy-MM-dd"),
+          params.CodiceFiscale,
+          params.FornituraNucleo,
+          params.CodiceSoglia
+        );
+        const response = await fetch(`${INPS_SERVICE_HOST}${endpoint}`, {
```

Comment:
use abortable fetch with timeout and http agent, see:

https://github.com/pagopa/io-functions-services/blob/0ee4ea8f7f85574ff0be0220666e031ba2407e1f/WebhookNotificationActivity/index.ts#L43

(create the fetch at top level since it reads process.env)

Reply:
`abortableFetch` use an https client. This change requires a change even on _io-bonus-inps-node-mock_?

Reply:
you can plug an http client as web using getHttpFetch

Reply:
i suggest to just copy and adapt the relevant parts from https://github.com/pagopa/io-ts-commons/blob/master/src/agent.ts

Reply:
use of abortableFetch in e70b96b